src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,11 +1047,28 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static SyntaxKind[] _lazyContextualKeywordKindsArray;
         public static IEnumerable<SyntaxKind> GetContextualKeywordKinds()
         {
-            for (int i = (int)SyntaxKind.YieldKeyword; i <= (int)SyntaxKind.WhenKeyword; i++)
+            if (_lazyContextualKeywordKindsArray == null)
             {
-                yield return (SyntaxKind)i;
+                ushort firstKeywordIdInclusive = (ushort)SyntaxKind.YieldKeyword;
```

Comment:
❓ Can you just keep these as type `SyntaxKind`, with the only cast being when you call `IsDefined`?